devdoc/tqueue_requirements.md

```diff
@@ -273,3 +274,21 @@ TQUEUE_POP_RESULT TQUEUE_POP(T)(TQUEUE(T) tqueue, T* item, void* pop_function_co
     - **SRS_TQUEUE_01_045: [** `TQUEUE_POP(T)` shall call `dispose_item_function` with `dispose_item_function_context` as `context` and the array entry value whose state was changed to `POPPING` as `item`. **]**
 
   - **SRS_TQUEUE_01_034: [** `TQUEUE_POP(T)` shall set the state to `NOT_USED` by using `interlocked_exchange`, succeed and return `TQUEUE_POP_OK`. **]**
+
+
+### TQUEUE_POP(T)
+```c
+int64_t TQUEUE_GET_COUNT(T)(TQUEUE(T) tqueue);
+```
+
+`TQUEUE_GET_COUNT(T)` returns the item count of the queue if available.
+
+**SRS_TQUEUE_22_001: [** If `tqueue` is `NULL` then `TQUEUE_GET_COUNT(T)` shall return zero. **]**
+
+**SRS_TQUEUE_22_002: [** `TQUEUE_GET_COUNT(T)` shall obtain the current head queue by calling `interlocked_add_64`. **]**
+
+**SRS_TQUEUE_22_003: [** `TQUEUE_GET_COUNT(T)` shall obtain the current tail queue by calling `interlocked_add_64`. **]**
+
+**SRS_TQUEUE_22_004: [** If the queue is empty (current tail >= current head), `TQUEUE_GET_COUNT(T)` shall return zero. **]**
```

Comment:
The condition for empty might be `current_head == (current_tail - 1)` if this is a circular queue.

Reply:
See SRS_TQUEUE_01_035.  head and tail are intialized to 0 when the queue was created (empty).